app/src/main/java/com/rkm/tasky/feature/agenda/model/AgendaItem.kt

```diff
@@ -0,0 +1,30 @@
+package com.rkm.tasky.feature.agenda.model
+
+sealed class AgendaItem {
+    abstract val id: String
+    abstract val title: String
+    abstract val description: String
+}
+
+data class AgendaTaskItem(
+    override val id: String,
+    override val title: String,
+    override val description: String,
+    val time: Long,
+    val isDone: Boolean
+): AgendaItem()
```

Comment:
Alternatively, you could think about creating a sealed hierarchy just for the differences of the agenda items. This is what I mean:
```
sealed interface AgendaItemDetails {
    data class Event(
        val toTime: LocalDateTime,
        val attendees: List<Attendee>,
        val photos: List<Photo>
    ): AgendaItemDetails
    
    data class Task(
        val isDone: Boolean
    ): AgendaItemDetails
    
    data object Reminder: AgendaItemDetails
}
```
That way, you just need type-checks when updating properties that are individual to one of the agenda item types, but you don't need this for updating shared properties anymore like the title:
```
data class AgendaDetailState(
    val title: String = "",
    val description: String = "",
    val time: LocalDateTime = LocalDateTime.now(),
    val details: AgendaItemDetails? = null
)
```
Updating then works like this:
```
// Updating a shared property
state = state.copy(title = "new title")

// Updating an individual property
state = state.copy(
    details = detailsAsEvent()?.copy(attendees = newAttendees)
)
```